src/script/lua_api/l_object.cpp

```diff
@@ -535,14 +542,137 @@ int ObjectRef::l_get_bone_position(lua_State *L)
 		return 0;
 
 	std::string bone = readParam<std::string>(L, 2, "");
+	BoneOverride props = sao->getBoneOverride(bone);
+	push_v3f(L, props.position.vector);
+	v3f euler_rot;
+	props.rotation.next.toEuler(euler_rot);
+	push_v3f(L, euler_rot * core::RADTODEG);
+	return 2;
+}
 
-	v3f position = v3f(0, 0, 0);
-	v3f rotation = v3f(0, 0, 0);
-	sao->getBonePosition(bone, &position, &rotation);
+// set_bone_override(self, bone, override)
+int ObjectRef::l_set_bone_override(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	ObjectRef *ref = checkObject<ObjectRef>(L, 1);
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == NULL)
+		return 0;
 
-	push_v3f(L, position);
-	push_v3f(L, rotation);
-	return 2;
+	std::string bone;
+	if (!lua_isnil(L, 2))
+		bone = readParam<std::string>(L, 2);
+
+	BoneOverride props;
+	if (lua_isnoneornil(L, 3)) {
+		sao->setBoneOverride(bone, props);
+		return 0;
+	}
+
+	auto read_prop_attrs = [L](auto &prop) {
+		lua_getfield(L, -1, "absolute");
+		prop.absolute = lua_toboolean(L, -1);
+		lua_pop(L, 1);
+
+		lua_getfield(L, -1, "interpolate");
+		if (lua_isnumber(L, -1))
+			prop.interp_timer = lua_tonumber(L, -1);
+		lua_pop(L, 1);
+	};
+
+	lua_getfield(L, 3, "position");
+	if (!lua_isnil(L, -1)) {
+		lua_getfield(L, -1, "vec");
+		props.position.vector = lua_isnil(L, -1) ? v3f() : check_v3f(L, -1);
```

Comment:
is there a reason for this pattern?
`props.position.vector` will be default initialized already

Reply:
Most probably copy-paste or I didn't think of the default initialization. Maybe I also found this more explicit / less error prone?
Note that in contrast to position, scale needs `v3f(1)`; relying on the implicit default-initialization for position & rotation (but not scale) may be a bit harder on the reader?

Either way I don't feel strongly about this, and I agree that dead code should be eliminated. I assume you would prefer `if (!lua_isnil(L, -1)) props.position.vector = check_v3f(L, -1);` (formatted properly)?

Reply:
Eliminated the unnecessary ternaries for now. Feel free to revert https://github.com/minetest/minetest/pull/12388/commits/61d765addb00839ca061c7c3bac315df87ea5dfd if you found the ternaries more readable.